### PR TITLE
Update miniconda from py38_4.8.3 to py38_4.9.2

### PR DESCRIPTION
**Pull request by rajivshah3**
Created with `brew bump-cask-pr`.

### CODE REVIEW

**Comment by rajivshah3** [comment]
I'm not sure why it's failing 🤔

**Comment by Aisuko** [comment]
Thanks for contributing and this PR can fix https://github.com/conda/conda/issues/9589

**Comment by rajivshah3** [comment]
> Could you try to trigger the CI job, I saw the failure message was `Error: It seems there is already a Binary at '/usr/local/bin/conda'.`

I previously force pushed to trigger it again since it ran into that error before, but it seems like it's still occurring. I wonder if it's conflicting with `anaconda`?

**Comment by Aisuko** [comment]
Hi @holymonson, I saw that you make the upgrade of  `miniconda` version succeed with the latest PR. Would you like to help us out here?

**Comment by rajivshah3** [comment]
In the meantime, I'll see if #97008 resolves the issue

**Comment by rajivshah3** [comment]
Hi @reitermarkus , sorry to ping you about this but I thought you might know the answer -- is it possible that the Miniconda from Homebrew is conflicting with the one [installed on the GitHub Actions runner](https://github.com/actions/virtual-environments/blob/64c9751269fda5e0d4a93407a59e0ed8c421ab87/images/macos/provision/core/miniconda.sh#L9)?

**Comment by Aisuko** [comment]
> In the meantime, I'll see if #97008 resolves the issue
> 
> Edit: That didn't help confused

I have no idea that the CI did not trigger the step `Uninstall conflicting formulae` or `brew uninstall --cask --zap miniconda`

**Comment by reitermarkus** [comment]
Is the `conda` on CI a symlink? If yes, what is it pointing to?

**Comment by Aisuko** [comment]
> Is the `conda` on CI a symlink? If yes, what is it pointing to?

See the steps in https://github.com/Homebrew/homebrew-cask/pull/96911/checks?check_run_id=1613209640

**Comment by rajivshah3** [comment]
> Is the `conda` on CI a symlink? If yes, what is it pointing to?

It seems like it. The script to provision the runner runs `sudo ln -s /usr/local/miniconda/bin/conda /usr/local/bin/conda`. `/usr/local/bin/conda` is the same path the tests say the conflicting binary is in.

**Comment by holymonson** [comment]
Would you comment this line and test if it could successfully install?

https://github.com/Homebrew/homebrew-cask/blob/f4b3e2db15a1a43c68c333954a1105b7e046089f/Casks/miniconda.rb#L17

File: Casks/miniconda.rb [hunk 1]
```diff
@@ -14,7 +14,7 @@
     executable: "Miniconda3-#{version}-MacOSX-x86_64.sh",
     args:       ["-b", "-p", "#{caskroom_path}/base"],
   }
-  binary "#{caskroom_path}/base/condabin/conda"
+  #binary "#{caskroom_path}/base/condabin/conda"
```
Comment (holymonson) [inline Casks/miniconda.rb:17]: I tested on my machine, this line seems no need commented. I don't know what's wrong with `/usr/local/bin/conda` on CI system

File: Casks/miniconda.rb [hunk 2]
```diff
@@ -1,6 +1,6 @@
 cask "miniconda" do
-  version "py38_4.8.3"
-  sha256 "9b9a353fadab6aa82ac0337c367c23ef842f97868dcbb2ff25ec3aa463afc871"
+  version "py38_4.9.2"
```
Comment (holymonson) [inline Casks/miniconda.rb:2]: `py39_4.9.2` is out too, update to py39?

Comment (holymonson) [reply Casks/miniconda.rb:2]: the latest is point to py38, perhaps py39 is unstable. Please ignore my previous comment.

File: Casks/miniconda.rb (hunk 1 above)
Comment (rajivshah3) [reply Casks/miniconda.rb:17]: I believe the issue is that Miniconda (not installed with Homebrew) is already [installed](https://github.com/actions/virtual-environments/blob/64c9751269fda5e0d4a93407a59e0ed8c421ab87/images/macos/provision/core/miniconda.sh#L9) in that path on GitHub Actions CI

**Comment by rajivshah3** [comment]
Aha, this happened with CocoaPods as well. The solution was to remove the symlink before running the rest of the CI script (#88948)

**Review by reitermarkus — APPROVED** [review]

**Pull request merged**